%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
% Welcome to Overleaf --- just edit your article on the left,
% and we'll compile it for you on the right. If you give 
% someone the link to this page, they can edit at the same
% time. See the help menu above for more info. Enjoy!
%
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
%
% For more detailed article preparation guidelines, please see:
% https://f1000research.com/for-authors/article-guidelines/software-tool-articles and http://f1000research.com/data-preparation

\documentclass[9pt,a4paper]{extarticle}
\usepackage{f1000_styles}
\usepackage{verbatim}
%% Default: numerical citations
\usepackage[numbers]{natbib}
\newenvironment{allintypewriter}{\ttfamily}{\par}

%% Uncomment this lines for superscript citations instead
% \usepackage[super]{natbib}

%% Uncomment these lines for author-year citations instead
% \usepackage[round]{natbib}
% \let\cite\citep

\begin{document}
\pagestyle{front}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\title{pysradb: A python package to query next-generation sequencing metadata
and data from NCBI Sequence Read Archive}
%\titlenote{The title should be detailed enough for someone to know whether the article would be of interest to them, but also concise. Please ensure the broadness and claims within the title are appropriate to the content of the article itself.}
\author[]{Saket Choudhary}
\affil[]{Computational Biology and Bioinformatics\\
University Of Southern California, Los Angeles, CA 900089, USA.}
\affil[]{email: skchoudh@usc.edu}


\maketitle
\thispagestyle{front}


\begin{abstract}
%Is the rationale for developing the new software tool clearly explained?
%Is the description of the software tool technically sound?
%Are sufficient details of the code, methods and analysis (if applicable) provided to allow replication of the software development and its use by others?
%Is sufficient information provided to allow interpretation of the expected output datasets and any results generated using the tool?
%Are the conclusions about the tool and its performance adequately supported by the findings presented in the article?

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
NCBI's Sequence Read Archive (SRA) is the primary archive of next-generation
sequencing datasets. SRA makes metadata and raw sequencing data available to the
research community to encourage reproducibility besides open avenues for testing
novel hypotheses on publicly available data. However, methods to programmatically
access this data are limited and not intuitive. We introduce \texttt{pysradb} package that provides a collection of simple command line methods to query and download metadata and data from SRA. It utilizes the curated metadata database available through the SRAdb project. We demonstrate the utility of \texttt{pysradb} for different use cases.

%Abstracts should be up to 300 words and provide a succinct summary of the article. Although the abstract should explain why the article might be interesting, care should be taken not to inappropriately over-emphasise the importance of the work described in the article. Citations should not be used in the abstract, and the use of abbreviations should be minimized.

\end{abstract}

\section*{Keywords}
bioinformatics, SRA, NGS, NCBI, metadata, GEO, python
%Please list up to eight keywords to help readers interested in your article find it more easily.



\clearpage
\pagestyle{main}
\section*{Introduction}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Several projects have made efforts to analyze and publish summaries of DNA-sequencing \cite{macarthur2012systematic} and RNA-sequencing \cite{lachmann2018massive, collado2017reproducible} datasets. Obtaining metadata and raw data from NCBI's Sequencing Read Archive (SRA)
\cite{leinonen2010sequence} are often the first steps towards re-analyzing public
next-generation sequencing datasets to compare them to private data or test a 
novel hypothesis. NCBI's SRA toolkit \cite{ncbisratoolit} provides utility methods
to download raw sequencing data, while the metadata can be obtained 
by querying the website or through the Entrez command line utility \cite{kans2018entrez}.


In order to make querying both metadata and data more precise and robust, SRAdb
\cite{zhu2013sradb} project provides a frequently updated SQLite database containing all the metadata parsed from SRA. \texttt{SRAdb} tracks the five main data objects in SRA's metadata: submission, study, sample, experiment and run. These are mapped to five different relational database tables that are made available in the SQLite file. SRAdb makes no attempts to either curate or clean up the metadata in any way. The metadata semantics remain same as they are in SRA with minor changes in the field names to imrove usability of SQL queries. The accompanying package made available in the R programming language, also called \texttt{SRAdb} \cite{zhudavissradb},
provides a convenient framework to handle metadata query and raw data downloads by 
utilizing the SQLite dabase. Though powerful, SRAdb's interface still requires the end user to be familiar with the R programming language. 


\texttt{pysradb} package builds up on the principles of \texttt{SRAdb} and 
provides a simple and intuitive interface for querying metadata and downloading
datasets from SRA through command line utility. It obviates the need for the
user to be famililar with any programming language as far as querying and downloading sequencing datasets is concerned. Additionally, it provides utility functions that will further help the user perform more granular queries, that
are often required while datasets at large scale.






\section*{Methods}

\subsection*{Implementation}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\texttt{pysradb} is implemented in Python (Python Software Foundation, https://www.python.org/) \cite{vanRossum:2011:PLR:2011965}
and uses \texttt{pandas} \cite{mckinney-proc-scipy-2010} for
data frame based operations. Since, downloading datasets can often take long time,
\texttt{pysradb} displays progress for long haul tasks using \texttt{tqdm} \cite{casper_da_costa_luis_2018_1211527}.
The metadata information is read in the form of a SQLite \cite{about_sqlite} database made available by SRAdb \cite{zhu2013sradb}. \texttt{pysradb} also supports accessing metadata 
information from Gene Expression Omnibus (GEO)
 \cite{edgar2002gene,barrett2012ncbi} through the SQLite database made available 
through the GEOmetadb project \cite{zhu2008geometadb}.

%For software tool papers, this section should address how the tool works and any relevant technical details required for implementation of the tool by other developers.  

\subsection*{Operation}
%This part of the methods should include the minimal system requirements needed to run the software and an overview of the workflow for the tool for users of the tool.

\texttt{pysradb} can be run on either Linux or Mac based operating systems. It is 
implemented in Python programming language, has minimal dependecies and can be easily installed using either \texttt{pip} or \texttt{conda} based package manager visa the \texttt{bioconda} \cite{gruning2018bioconda} channel. It works both in Python 2 and Python 3 enviroments.

\begin{comment}


\section*{Results} % Optional - only if novel data or analyses are included
This section is only required if the paper includes novel data or analyses, and should be written as a traditional results section.
\end{comment}

\section*{Use Cases} % Optional - only if NO new datasets are included
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
The primary use case of \texttt{pysradb} is in automated download of an entire
SRA project. NCBI's sratoolkit \cite{ncbisratoolit}
This section is required if the paper does not include novel data or analyses. 
Examples of input and output files should be provided with some explanatory context.  Any novel or complex variable parameters should also be explained in sufficient detail to allow users to understand and use the tool's functionality.


\textbf{Step 1: }

\begin{allintypewriter}
\$ pysradb srametadb

Downloading SRAmetadb.sqlite.gz: 2.15GB [01:22, 28.0MB/s]   

Extracting data/SRAmetadb.sqlite.gz ...

Extracting SRAmetadb.sqlite.gz: 33.0GB [07:51, 75.2MB/s]

Done!

Metadata associated with data/SRAmetadb.sqlite:

                 name                value

0      schema version                  1.0

1  creation timestamp  2018-12-07 00:39:29

\end{allintypewriter}

\textbf{Step 2:}

\begin{allintypewriter}
\$ pysradb sra-metadata --db data/SRAmetadb.sqlite SRP098789 


\end{allintypewriter}

The simplest use case of `pysradb` is when you apriopri know the SRA project ID (SRP)
and would simply want to fetch the metadata associated with it. This is generally
reflected in the `SraRunTable.txt' that you get from NCBI's website.
Example: https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP098789.


\begin{allintypewriter}
pysradb sra-metadata --db data/SRAmetadb.sqlite SRP098789
\end{allintypewriter}


Once you have fetched the metadata and made sure, this is the project
you were looking for, you would want to download everything at once.
NCBI follows this hiererachy: `SRP => SRX => SRR`. Each `SRP` (project) has multiple
`SRX` (experiments) and each `SRX` in turn has multiple `SRR` (runs) inside it.
We want to mimick this hiereachy in our downloads. The reason to do that is simple:
in most cases you care about `SRX` the most, and would want to "merge" your SRRs
in one way or the other. Having this hierearchy ensures your downstream code
can handle such cases easily, without worrying about which runs (SRR) need to be merged.


\begin{allintypewriter}
!pysradb download --db data/SRAmetadb.sqlite SRP063852 
\end{allintypewriter}

Often, you need to process only a smaller set of samples from a project (SRP).
Consider this project which has data spanning four assays.



But, you might be only interested in analyzing the `RNA-seq` samples and would just want to download that subset.
This is simple using `pysradb` since the metadata can be subset just as you would subset a dataframe in
pandas.




\begin{allintypewriter}
!pysradb sra-metadata --db data/SRAmetadb.sqlite  SRP098789 --expand
\end{allintypewriter}



Another common operation that we do on SRA is seach, plain text search.


If you want to look up for all projects where `ribosome profiling` appears somewhere
in the description:


\begin{comment}

\section*{Discussion} % Optional - only if novel data or analyses are included
This section is only required if the paper includes novel data or analyses, and should be written in the same style as a traditional discussion section.
Please include a brief discussion of allowances made (if any) for controlling bias or unwanted sources of variability, and the limitations of any novel datasets.


\section*{Conclusions} % Optional - only if novel data or analyses are included
This section is only required if the paper includes novel data or analyses, and should be written as a traditional conclusion.

\section*{Summary} % Optional - only if NO new datasets are included
This section is required if the paper does not include novel data or analyses.  It allows authors to briefly summarize the key points from the article.


\end{comment}

\section*{Data availability} % Optional - only if novel data or analyses are included
Please add details of where any datasets that are mentioned in the paper, and that have not have not previously been formally published, can be found.  If previously published datasets are mentioned, these should be cited in the references, as per usual scholarly conventions.

\section*{Software availability}
Software and source code available from: https://github.com/saketkc/pysradb

Documentation available at: https://saketkc.github.io/pysradb

Archived source code at time of publication: 

Software license: BSD-3-Clause

\section*{Author Contributions}
S.C. designed the project, implemented the package, and wrote the manuscript.

\section*{Competing interests}
No competing interests were disclosed.

\section*{Grant information}
The author declared that no grants were involved in supporting this work.

\section*{Acknowledgments}
\begin{comment}
This section should acknowledge anyone who contributed to the research or the
article but who does not qualify as an author based on the criteria provided earlier
(e.g. someone or an organization that provided writing assistance). Please state how
they contributed; authors should obtain permission to acknowledge from all those
mentioned in the Acknowledgments section.

Please do not list grant funding in this section.
\end{comment}

{\small\bibliographystyle{unsrtnat}
\bibliography{bibliography}}



% See this guide for more information on BibTeX:
% http://libguides.mit.edu/content.php?pid=55482&sid=406343

% For more author guidance please see:
% https://f1000research.com/for-authors/article-guidelines/software-tool-articles


% When all authors are happy with the paper, use the 
% ‘Submit to F1000RESEARCH' button from the menu above
% to submit directly to the open life science journal F1000Research.

% Please note that this template results in a draft pre-submission PDF document.
% Articles will be professionally typeset when accepted for publication.

% We hope you find the F1000Research Overleaf template useful,
% please let us know if you have any feedback using the help menu above.


\end{document}