\documentclass[9pt,a4paper]{extarticle}
\usepackage[T1]{fontenc}
\usepackage{tabto}
\usepackage{float}
\catcode`\_=12

\usepackage{f1000_styles}
\usepackage{verbatim}
%% Default: numerical citations
\usepackage[numbers]{natbib}
%% Uncomment this lines for superscript citations instead
% \usepackage[super]{natbib}
%% Uncomment these lines for author-year citations instead
% \usepackage[round]{natbib}
% \let\cite\citep

\newenvironment{allintypewriter}{\ttfamily}{\par}



\begin{document}
\pagestyle{front}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\title{pysradb: A python package to query next-generation sequencing metadata
and data from NCBI Sequence Read Archive}
%\titlenote{The title should be detailed enough for someone to know whether the
%article would be of interest to them, but also concise. Please ensure the 
%broadness and claims within the title are appropriate to the content of the
%article itself.}

\author[]{Author}%Saket Choudhary}
\affil[]{Computational Biology and Bioinformatics\\
University Of Southern California, Los Angeles, CA 900089, USA.}
\affil[]{email: }


\maketitle
\thispagestyle{front}


\begin{abstract}
%Is the rationale for developing the new software tool clearly explained?

%Is the description of the software tool technically sound?

%Are sufficient details of the code, methods and analysis (if applicable) provided to allow replication of the software development and its use by others?

%Is sufficient information provided to allow interpretation of the expected output datasets and any results generated using the tool?

%Are the conclusions about the tool and its performance adequately supported by the findings presented in the article?

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
NCBI’s Sequence Read Archive (SRA) is the primary archive of next-generation 
sequencing datasets. SRA makes metadata and raw sequencing data available to the
research community to encourage reproducibility, and to provide avenues for 
testing novel hypotheses on publicly available data. However, methods to
programmatically access this data are limited and not intuitive. We introduce a
Python package \texttt{pysradb} that provides a collection of simple command
line methods to query and download metadata and data from SRA. It utilizes the
curated metadata database available through the SRAdb project. We demonstrate 
the utility of pysradb for different use cases. 
%\texttt{pysradb} streamlines the process of retriving data and metadata

%Abstracts should be up to 300 words and provide a succinct summary of the
%article. Although the abstract should explain why the article might be 
% interesting, care should be taken not to inappropriately over-emphasise the 
%importance of the work described in the article. Citations should not be used
%in the abstract, and the use of abbreviations should be minimized.

\end{abstract}

\section*{Keywords}
bioinformatics, SRA, NGS, NCBI, metadata, GEO, python
%Please list up to eight keywords to help readers interested in your article find it more easily.



\clearpage
\pagestyle{main}
\section*{Introduction}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Several projects have made efforts to analyze and publish summaries of DNA-
sequencing \cite{macarthur2012systematic} and RNA-sequencing
\cite{lachmann2018massive, collado2017reproducible} datasets. Obtaining metadata
and raw data from NCBI's Sequencing Read Archive (SRA) \cite{leinonen2010sequence}
are often the first steps towards re-analyzing public next-generation sequencing 
datasets to compare them to private data or test a novel hypothesis. NCBI's SRA
toolkit \cite{ncbisratoolit} provides utility methods to download raw sequencing
data, while the metadata can be obtained by querying the website or through the
Entrez command line utility 
\cite{kans2018entrez}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
In order to make querying both metadata and data more precise and robust, SRAdb
\cite{zhu2013sradb} project provides a frequently updated SQLite database 
containing all the metadata parsed from SRA. \texttt{SRAdb} tracks the five main
data objects in SRA's metadata: submission, study, sample, experiment and run.
These are mapped to five different relational database tables that are made
available in the SQLite file. SRAdb's metadata semantics remain as they are in
SRA with minor changes in the field names to improve usability of SQL queries. 
The accompanying package made available in the R programming language, also
called \texttt{SRAdb} \cite{zhudavissradb}, provides a convenient framework to
handle metadata query and raw data downloads by utilizing the SQLite dabase. 
Though powerful, SRAdb's interface still requires the end user to be familiar 
with the R programming language. 

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\texttt{pysradb} package builds up on the principles of \texttt{SRAdb} and 
provides a simple and intuitive interface for querying metadata and downloading
datasets from SRA through command line utility. It obviates the need for the
user to be familiar with any programming language as far as querying and
downloading sequencing datasets is concerned. Additionally, it provides utility 
functions that will further help the user perform more granular queries, that
are often required with datasets at large scale.






\section*{Methods}

\subsection*{Implementation}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\texttt{pysradb} is implemented in Python (Python Software Foundation, 
https://www.python.org/) \cite{vanRossum:2011:PLR:2011965} and uses
\texttt{pandas} \cite{mckinney-proc-scipy-2010} for data frame based operations. 
Since, downloading datasets can often take long time, \texttt{pysradb} displays 
progress for long haul tasks using \texttt{tqdm}
\cite{casper_da_costa_luis_2018_1211527}.The metadata information is read in the
form of a SQLite \cite{about_sqlite} database made available by SRAdb
\cite{zhu2013sradb}. \texttt{pysradb} also supports accessing metadata 
information from Gene Expression Omnibus (GEO) \cite{edgar2002gene,
barrett2012ncbi} through the SQLite database made available through the
GEOmetadb project \cite{zhu2008geometadb}.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\texttt{pysradb} can be run on either Linux or Mac based operating systems. It
is implemented in Python programming language, has minimal dependencies and can
be easily installed using either \texttt{pip} or \texttt{conda} based package
manager visa the \texttt{bioconda} \cite{gruning2018bioconda} channel. It works
both in Python 2 and Python 3 environments.

%For software tool papers, this section should address how the tool works and any relevant technical details required for implementation of the tool by other developers.  

\subsection*{Operation}
%This part of the methods should include the minimal system requirements needed to run the software and an overview of the workflow for the tool for users of the tool.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\texttt{pysradb} uses SQLite file produced and made available by SRAdb 
\cite{zhu2013sradb} project. The file itself can be downloaded using
\texttt{pysradb} as:

\begin{allintypewriter}
\$ pysradb srametadb

Downloading SRAmetadb.sqlite.gz: 2.15GB [01:22, 28.0MB/s]   

Extracting data/SRAmetadb.sqlite.gz ...

Extracting SRAmetadb.sqlite.gz: 33.0GB [07:51, 75.2MB/s]

Done!

Metadata associated with data/SRAmetadb.sqlite:

                 name                value

0      schema version                  1.0

1  creation timestamp  2018-12-07 00:39:29

\end{allintypewriter}

SRAmetadb.sqlite file is required for all other operations supported
by \texttt{pysradb}.


\section*{Use Cases} % Optional - only if NO new datasets are included
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
The primary use case of \texttt{pysradb} is in automated download of an entire
SRA project. NCBI's sratoolkit \cite{ncbisratoolit} allows downloading fetching 
raw data per sequencing run. Each SRA project (SRP) consists of one or multiple 
experiments (SRX) which are sequenced as one or multiple runs (SRR). 
\texttt{sratoolkit} allows downloading 


\subsection*{Getting a list of GEO experiments for a GEO study}
Any GEO study (GSE) will involve a collection of experiments (GSM). We can obtain
an entire list of experiments corresponding to the study using the
\texttt{gse-to-gsm} subcommand from \texttt{pysradb}:

\begin{allintypewriter}
\$ pysradb gse-to-gsm GSE41637 | head\\
\begin{table}[H]
    \begin{tabular}{cc}
        study_alias & experiment_alias\\
        GSE41637  &  GSM1020640_1\\
        GSE41637  &  GSM1020641_1\\
        GSE41637  &  GSM1020642_1\\
        GSE41637  &  GSM1020643_1\\
        GSE41637  &  GSM1020644_1\\
        GSE41637  &  GSM1020645_1\\
        GSE41637  &  GSM1020646_1\\
        GSE41637  &  GSM1020647_1\\
        GSE41637  &  GSM1020648_1\\ 
    \end{tabular}
\end{table}

\end{allintypewriter}

However, just a list of GSM id is not useful if one is performing any downstream
analysis which essentially requires more detailed information about the metadata
associated with each experiment. This relevant metadata associated with each sample can be obtained by provinding \texttt{gse-to-gsm} additional flags:

\begin{allintypewriter}
\$ pysradb gse-to-gsm --desc GSE41637 | head\\
\begin{table}[H]
    \begin{tabular}{lll}
        study_alias & experiment_alias & sample_attribute\\
        GSE41637 & GSM1020640_1 &  source_name: mouse_brain || strain: DBA/2J || tissue: brain\\
        GSE41637 & GSM1020641_1 &  source_name: mouse_colon || strain: DBA/2J || tissue: colon\\
        GSE41637 & GSM1020642_1 &  source_name: mouse_heart || strain: DBA/2J || tissue: heart\\
        GSE41637 & GSM1020643_1 &  source_name: mouse_kidney || strain: DBA/2J || tissue: kidney\\
        GSE41637 & GSM1020644_1 &  source_name: mouse_liver || strain: DBA/2J || tissue: liver\\
        GSE41637 & GSM1020645_1 &  source_name: mouse_lung || strain: DBA/2J || tissue: lung\\
        GSE41637 & GSM1020646_1 &  source_name: mouse_skm || strain: DBA/2J || tissue: skeletal muscle\\
        GSE41637 & GSM1020647_1 &  source_name: mouse_spleen || strain: DBA/2J || tissue: spleen\\
        GSE41637 & GSM1020648_1 &  source_name: mouse_testes || strain: DBA/2J || tissue: testes\\
    \end{tabular}
\end{table}
\end{allintypewriter}

The metadata information can then be parsed from the \texttt{sample\_attribute} column.
To obtain a more strucutred metadata, we can use an additional flag \texttt{--expand}:

\begin{allintypewriter}
\$ pysradb gse-to-gsm --desc --expand GSE41637 | head\\
\begin{table}[H]
    \begin{tabular}{lllll}
        study_alias & experiment_alias & source_name &  strain & tissue\\
        GSE41637 & GSM1020640_1 &  mouse_brain &  dba/2j & brain\\
        GSE41637 & GSM1020641_1 &  mouse_colon &  dba/2j & colon\\
        GSE41637 & GSM1020642_1 &  mouse_heart &  dba/2j & heart\\
        GSE41637 & GSM1020643_1 &  mouse_kidney & dba/2j & kidney\\
        GSE41637 & GSM1020644_1 &  mouse_liver &  dba/2j & liver\\
        GSE41637 & GSM1020645_1 &  mouse_lung & dba/2j & lung\\
        GSE41637 & GSM1020646_1 &  mouse_skm & dba/2j & skeletal muscle\\
    \end{tabular}
\end{table}
\end{allintypewriter}

%% pysradb sra-metadata --desc --expand SRP109126 | grep 'study_accession\|ribosome protected footprints' | pysradb download


\subsection*{Getting SRP from GSE}
\begin{allintypewriter}
\$ pysradb gse-to-srp --desc --expand GSE100007
\begin{table}[H]
    \begin{tabular}{llllll}
study_alias & study_accession & cell_type & fraction & molecule & source_name\\                       GSE100007 &  SRP109126   &    human embryonic stem cells (hesc) & cytoplasm                      & cytoplasmic rna               &  human embryonic stem cells (hesc)\\
GSE100007 &  SRP109126   &    human embryonic stem cells (hesc) & high polysome (5-8+ ribosomes) & polysomal rna                 &  human embryonic stem cells (hesc)\\
GSE100007 &  SRP109126   &    human embryonic stem cells (hesc) & low polysome (2-4 ribosomes)   & polysomal rna                 &  human embryonic stem cells (hesc)\\
GSE100007 &  SRP109126   &    human embryonic stem cells (hesc) & monosome (80s)                 & monosomal rna                 &  human embryonic stem cells (hesc)\\
GSE100007 &  SRP109126   &    human embryonic stem cells (hesc) & nucleus                        & nuclear rna                   &  human embryonic stem cells (hesc)\\
GSE100007 &  SRP109126   &    human embryonic stem cells (hesc) & ribosome protected footprints  & ribosome protected footprints &  human embryonic stem cells (hesc)\\
GSE100007 &  SRP109126   &    neural cultures (14 days)         & cytoplasm                      & cytoplasmic rna               &  neural cultures (14 days)\\
GSE100007 &  SRP109126   &    neural cultures (14 days)         & high polysome (5-8+ ribosomes) & polysomal rna                 &  neural cultures (14 days)\\
GSE100007 &  SRP109126   &    neural cultures (14 days)         & low polysome (2-4 ribosomes)   & polysomal rna                 &  neural cultures (14 days)\\
GSE100007 &  SRP109126   &    neural cultures (14 days)         & monosome (80s)                 & monosomal rna                 &  neural cultures (14 days)\\
GSE100007 &  SRP109126   &    neural cultures (14 days)         & nucleus                        & nuclear rna                   &  neural cultures (14 days)\\
GSE100007 &  SRP109126   &    neural cultures (14 days)         & ribosome protected footprints  & ribosome protected footprints &  neural cultures (14 days)\\
GSE100007 &  SRP109126   &    neural cultures (50 days)         & cytoplasm                      & cytoplasmic rna               &  neural cultures (50 days)\\
GSE100007 &  SRP109126   &    neural cultures (50 days)         & high polysome (5-8+ ribosomes) & polysomal rna                 &  neural cultures (50 days)\\
GSE100007 &  SRP109126   &    neural cultures (50 days)         & low polysome (2-4 ribosomes)   & polysomal rna                 &  neural cultures (50 days)\\
GSE100007 &  SRP109126   &    neural cultures (50 days)         & monosome (80s)                 & monosomal rna                 &  neural cultures (50 days)\\
GSE100007 &  SRP109126   &    neural cultures (50 days)         & nucleus                        & nuclear rna                   &  neural cultures (50 days)\\
GSE100007 &  SRP109126   &    neural progenitor cells (npc)     & cytoplasm                      & cytoplasmic rna               &  neural progenitor cells (npc)\\
GSE100007 &  SRP109126   &    neural progenitor cells (npc)     & high polysome (5-8+ ribosomes) & polysomal rna                 &  neural progenitor cells (npc)\\
GSE100007 &  SRP109126   &    neural progenitor cells (npc)     & low polysome (2-4 ribosomes)   & polysomal rna                 &  neural progenitor cells (npc)\\
GSE100007 &  SRP109126   &    neural progenitor cells (npc)     & monosome (80s)                 & monosomal rna                 &  neural progenitor cells (npc)\\
GSE100007 &  SRP109126   &    neural progenitor cells (npc)     & nucleus                        & nuclear rna                   &  neural progenitor cells (npc)\\
GSE100007 &  SRP109126   &    neural progenitor cells (npc)     & ribosome protected footprints  & ribosome protected footprints &  neural progenitor cells (npc)\\
    \end{tabular}
\end{table}
\end{allintypewriter}

\subsection*{Getting SRPs from GSE}
\begin{allintypewriter}
\$ pysradb gse-to-srp GSE24355 GSE25842
\begin{table}[H]
    \begin{tabular}{ll}
    study_alias & study_accession\\
    GSE24355 & SRP003870\\
    GSE25842 & SRP005378\\
    \end{tabular}
\end{table}
\end{allintypewriter}

\subsection*{Getting project's metadata}
\begin{allintypewriter}
\$ pysradb metadata SRP026005 --assay --desc --expand  | head
\footnotesize{
\begin{table}[H]
    \begin{tabular}{llllllll}
study_accession & experiment_accession & sample_accession & run_accession & library_strategy & embryonic_stage & source_name &  strain  \\
SRP026005 &   SRX305245 & SRS444476 & SRR900108 & RNA-Seq &   e9.5 & neural crest cells sorted from embryo culture & c57bl/6\\
SRP026005 &   SRX305246 & SRS444467 & SRR900109 & RNA-Seq &   e9.5 & neural crest cells sorted from embryo culture & c57bl/6\\
SRP026005 &   SRX305247 & SRS444468 & SRR900110 & RNA-Seq &   e9.5 & neural crest cells sorted from embryo culture & c57bl/6\\
SRP026005 &   SRX305247 & SRS444468 & SRR900111 & RNA-Seq &   e9.5 & neural crest cells sorted from embryo culture & c57bl/6\\
SRP026005 &   SRX305248 & SRS444470 & SRR900112 & RNA-Seq &   e9.5 & neural crest cells sorted from embryo culture & c57bl/6\\
SRP026005 &   SRX305249 & SRS444471 & SRR900113 & RNA-Seq &   e9.5 & neural crest cells sorted from embryo culture & c57bl/6\\
SRP026005 &   SRX305250 & SRS444472 & SRR900114 & RNA-Seq &   e9.5 & neural crest cells sorted from embryo culture & c57bl/6\\
SRP026005 &   SRX305250 & SRS444472 & SRR900115 & RNA-Seq &   e9.5 & neural crest cells sorted from embryo culture & c57bl/6\\
SRP026005 &   SRX305251 & SRS444473 & SRR900116 & RNA-Seq &   e9.5 & neural crest cells sorted from embryo culture & c57bl/6\\
\end{tabular}
\end{table}
}
\end{allintypewriter}

\subsection*{Getting assay summary statistics}

\begin{allintypewriter}
\$ pysradb metadata SRP000941 --assay  | tr -s '  ' | cut -f5 -d ' ' | tail -n +2 | sort | uniq -c
\begin{table}[H]
    \begin{tabular}{ll}
        999  & Bisulfite-Seq\\
        768 & ChIP-Seq\\
        121 & OTHER\\
        353 & RNA-Seq\\
        28 & WGS\\
    \end{tabular}
\end{table}
\end{allintypewriter}

\subsection*{Downloading data}

\begin{allintypewriter}
\$  pysradb download -p SRP003870 -p SRP005378
\end{allintypewriter}







%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
The simplest use case of `pysradb` is when you apriopri know the SRA project ID 
(SRP) and would simply want to fetch the metadata associated with it. This is
generally reflected in the `SraRunTable.txt' that you get from NCBI's website.
Example: https://www.ncbi.nlm.nih.gov/Traces/study/?acc=SRP098789.

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{allintypewriter}
pysradb metadata SRP098789
\end{allintypewriter}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Once you have fetched the metadata and made sure, this is the project you were 
looking for, you would want to download everything at once. NCBI follows this 
hiererachy: `SRP => SRX => SRR`. Each `SRP` (project) has multiple `SRX` 
(experiments) and each `SRX` in turn has multiple `SRR` (runs) inside it.
We want to mimick this hiereachy in our downloads. The reason to do that is simple:
in most cases you care about `SRX` the most, and would want to "merge" your SRRs
in one way or the other. Having this hierearchy ensures your downstream code
can handle such cases easily, without worrying about which runs (SRR) need to
be merged.


\begin{allintypewriter}
\$ pysradb download -p SRP063852 
\end{allintypewriter}

Often, you need to process only a smaller set of samples from a project (SRP).
Consider the project SRP000941 which has data spanning four assays. But, we might be 
only interested in analyzing the `RNA-seq` samples and would just want to download
that subset. This can be done simply using the following command:



%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\begin{allintypewriter}
\$ pysradb metadata SRP000941 --assay $|$ grep `study$|$RNA-Seq' $|$ pysradb download
\end{allintypewriter}

\subsection*{Search}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
Another common operation that we do on SRA is search, plain text search. If we want
to look up for all projects where `ribosome profiling` appears somewhere
in the description, we can use:

\begin{allintypewriter}
\$ pysradb search '"ribosome profiling"'
\end{allintypewriter}



\begin{comment}

\section*{Discussion} % Optional - only if novel data or analyses are included
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
This section is only required if the paper includes novel data or analyses, and 
should be written in the same style as a traditional discussion section. Please 
include a brief discussion of allowances made (if any) for controlling bias or unwanted sources of variability, and the limitations of any novel datasets.


\section*{Conclusions} % Optional - only if novel data or analyses are included
This section is only required if the paper includes novel data or analyses, and should be written as a traditional conclusion.

\section*{Summary} % Optional - only if NO new datasets are included
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%% 
This section is required if the paper does not include novel data or analyses.  It allows authors to briefly summarize the key points from the article.
The need to perform metaanalysis warrants need of more stream lined ways to obtain
data. NCBI provides access to a huge set of datasets across 


\end{comment}

\section*{Data availability} % Optional - only if novel data or analyses are included
Please add details of where any datasets that are mentioned in the paper, and that have not have not previously been formally published, can be found.  If previously published datasets are mentioned, these should be cited in the references, as per usual scholarly conventions.

\section*{Software availability}
Software and source code available from: https://github.com/saketkc/pysradb

Documentation available at: https://saketkc.github.io/pysradb

Archived source code at time of publication: 

Software license: BSD-3-Clause

\section*{Author Contributions}
%S.C. designed the project, implemented the package, and wrote the manuscript.

\section*{Competing interests}
No competing interests were disclosed.

\section*{Grant information}
The author declared that no grants were involved in supporting this work.

\section*{Acknowledgments}
%We thank Shweta Ramdas (Universsity of Pennsylvania) for reviewing the earlier drafts.
%S.C. acknowledges support from USC Provost Graduate Research Fellowship. 

\begin{comment}
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
This section should acknowledge anyone who contributed to the research or the
article but who does not qualify as an author based on the criteria provided earlier
(e.g. someone or an organization that provided writing assistance). Please state how
they contributed; authors should obtain permission to acknowledge from all those
mentioned in the Acknowledgments section.

Please do not list grant funding in this section.
\end{comment}

{\small\bibliographystyle{unsrtnat}
\bibliography{bibliography}}

\begin{comment}

5. Main Body
Software Tool Articles typically contain the following sections:
Introduction
Methods, providing details of Implementation and Operation
Results (optional)
Use Cases (optional)
Discussion/Conclusions
The Introduction should provide context as to why the software tool was developed and what need it addresses. It is good scholarly practice to mention previously developed tools that address similar needs, and why the current tool is needed.
The Methods should include a subsection on Implementation describing how the tool works and any relevant technical details required for implementation; and a subsection on Operation, which should include the minimal system requirements needed to run the software and an overview of the workflow.
A Results section is only required if the paper includes novel data or analyses, and should be written as a traditional results section.
Please include a section on Use Cases if the paper does not include novel data or analyses. Examples of input and output files should be provided with some explanatory context. Any novel or complex variable parameters should be explained in sufficient detail to enable users to understand and use the tool's functionality.
A Discussion (e.g. if the paper includes novel data or analyses) or Conclusions should include a brief discussion of allowances made (if any) for controlling bias or unwanted sources of variability, and the limitations of any novel datasets.
Reproducibility: F1000Research is committed to serving the research community by ensuring that all articles include sufficient information to allow others to reproduce the work. With this in mind, Methods sections should provide sufficient details of the materials and methods used so that the work can be repeated by others. The section should also include a brief discussion of allowances made (if any) for controlling bias or unwanted sources of variability. Any limitations of the datasets should be discussed.
When antibodies are used, the species in which the antibody was raised, the manufacturing company or source laboratory, the catalogue or reference number, and whether it is a polyclonal or monoclonal antibody should be included. In addition, if the antibody has been previously validated, a reference to the validation study should be included. If the antibody has not been validated, full details of the dilution and use of the antibody should be given in the Methods section.
We encourage authors to add Research Resource Identifiers (RRIDs) to their article in order to unambiguously identify the following types of resources: antibodies, genetically modified organisms, software tools, data, databases and services. More information on this project is available from the Resource Identification Initiative and RRIDs can be obtained from the portal.
Where applicable, we also encourage authors to deposit a step-by-step description of their protocols on protocols.io, where they obtain a persistent digital object identifier (DOI), which can be included in the Methods section of the article, using https://doi.org/10.17504/protocols.io.[PROTOCOL DOI] as the format (e.g. https://doi.org/10.17504/protocols.io.hrkb54w). Authors should note that the protocol is only made public once they select “Publish” on protocols.io.

\end{comment}

% See this guide for more information on BibTeX:
% http://libguides.mit.edu/content.php?pid=55482&sid=406343

% For more author guidance please see:
% https://f1000research.com/for-authors/article-guidelines/software-tool-articles


% When all authors are happy with the paper, use the 
% ‘Submit to F1000RESEARCH' button from the menu above
% to submit directly to the open life science journal F1000Research.

% Please note that this template results in a draft pre-submission PDF document.
% Articles will be professionally typeset when accepted for publication.

% We hope you find the F1000Research Overleaf template useful,
% please let us know if you have any feedback using the help menu above.


\end{document}